### PR TITLE
Scala 2.13, SBT 1.5.7, Play 2.8 (and related upgrades)

### PR DESCRIPTION
**Pull request by nicl**
**Note, I have disabled compilation warnings for now (there remain a lot, mostly due to deprecations), as I plan on tackling these in a follow up PR.**

**Note, depends on https://github.com/guardian/riff-raff/pull/667 (now live).**

## What does this change?
Scala -> 2.13 (previously 2.12)
Play  -> 2.8 (2.6)
SBT   -> 1.5.6 (0.13)

This was a bit of a gigantic upgrade due to dependency binary
incompatibilities (dependency coupling) and general breaking
changes.

Hopefully in future we can do smaller and more frequent updates.

To break down the changes:

### 1. Removal of some niche dependencies

The list is: diffson-play-json, scala-ssh, henkan-convert.

These were either unused or used only a little.

### 2. .mapValues -> .view.mapValues(...).toMap

https://docs.scala-lang.org/overviews/core/collections-migration-213.html#what-are-the-breaking-changes

### 3. other Scala 2.12 and 2.13 changes

* .par now requires a special import
* Seq is now core.immutable.Seq
* import precendence and shadowing seems to have changed (see:
  scala/scala#6589), which I *think* caused this issue
  as task and deployment_types packages share many names
* requirement to drop parens when a method (typically controller
  method) was defined without them
* etc.

### 4. library breaking changes

For example, in the Guardian's own Play Secret and GoogleAuth
libraries, or in the Boostrap ('b3') library, especially in
how forms are defined.

### 5. SBT changes

The deprecation of 'in' for '/' instead.

## How to test

CI, and then a manual test of key functionality in CODE.

### CODE REVIEW

File: magenta-lib/src/main/scala/magenta/input/resolver/DeploymentTypeResolver.scala [hunk 1]
```diff
@@ -23,9 +23,18 @@ object DeploymentTypeResolver {
...
+          name = deployment.name,
+          `type` = deployment.`type`,
+          stacks = deployment.stacks,
+          regions = deployment.regions,
+          actions = as,
+          app = deployment.app,
+          contentDirectory = deployment.contentDirectory,
+          dependencies = deployment.dependencies,
+          parameters = deployment.parameters,
+        )
```
Comment (akash1810) [inline magenta-lib/src/main/scala/magenta/input/resolver/DeploymentTypeResolver.scala:37]: Is this the result of the removal of `henkan.convert.Syntax._`?

File: project/build.properties [hunk 2]
```diff
@@ -1 +1 @@
-sbt.version=0.13.18
+sbt.version=1.5.6
```
Comment (akash1810) [inline project/build.properties:1]: Should this be 1.5.7?

File: riff-raff/app/AppComponents.scala [hunk 3]
```diff
@@ -51,24 +51,18 @@ class AppComponents(context: Context, config: Config, passwordProvider: Password
...
-        overlapDuration = Duration.ofHours(2)
-      ),
-      parameterName = config.auth.secretStateSupplierKeyName,
-      ssmClient = AWSSimpleSystemsManagementClientBuilder.standard()
-        .withRegion(config.auth.secretStateSupplierRegion)
-        .withCredentials(config.credentialsProviderChainV1(None, None))
-        .build()
+    new parameterstore.SecretSupplier(
+      TransitionTiming(usageDelay = ofMinutes(3), overlapDuration = ofHours(2)),
+      "/Example/PlayAppSecret",
```
Comment (akash1810) [inline riff-raff/app/AppComponents.scala:56]: Was this change intentional?

File: magenta-lib/src/main/scala/magenta/input/resolver/DeploymentTypeResolver.scala (hunk 1 above)
Comment (nicl) [reply magenta-lib/src/main/scala/magenta/input/resolver/DeploymentTypeResolver.scala:37]: Yep. We only had a couple of use cases, so it seemed better to just write it out. In general, because of Scala's tendency for difficult upgrades, I recommend only adding third-party dependencies when there is a large and clear benefit.

File: project/build.properties (hunk 2 above)
Comment (nicl) [reply project/build.properties:1]: Ah yes, I think when I started on this 1.5.7 wasn't out. Will bump. Thanks.

File: riff-raff/app/AppComponents.scala (hunk 3 above)
Comment (nicl) [reply riff-raff/app/AppComponents.scala:56]: Doh, no. Let me fix that!

**Review by akash1810 — APPROVED** [review]
img

**Comment by akash1810** [comment]
This caused a couple of scheduled deploys to fail so we rolled back (not reverted yet though):
  - https://riffraff.gutools.co.uk/deployment/view/642384a0-5308-431f-8235-9346e424eec0
  - https://riffraff.gutools.co.uk/deployment/view/4d90a389-2a98-4e4e-9b23-38e3efcbcceb
  - https://riffraff.gutools.co.uk/deployment/view/3f78fe27-8724-4d5d-967f-9d8fac7b39ae
  - https://riffraff.gutools.co.uk/deployment/view/6ece1e05-15c8-46db-9be1-eaa2fcd0ad30

The error seen is:

```
software.amazon.awssdk.core.exception.SdkClientException Unable to marshall request to JSON: Can not write a field name, expecting a value
```

Full stack trace:

```
software.amazon.awssdk.core.exception.SdkClientException$BuilderImpl.build(SdkClientException.java:98)
software.amazon.awssdk.services.lambda.transform.UpdateFunctionCodeRequestMarshaller.marshall(UpdateFunctionCodeRequestMarshaller.java:52)
software.amazon.awssdk.services.lambda.transform.UpdateFunctionCodeRequestMarshaller.marshall(UpdateFunctionCodeRequestMarshaller.java:31)
software.amazon.awssdk.core.internal.handler.BaseClientHandler.lambda$finalizeSdkHttpFullRequest$0(BaseClientHandler.java:67)
software.amazon.awssdk.core.internal.util.MetricUtils.measureDuration(MetricUtils.java:51)
software.amazon.awssdk.core.internal.handler.BaseClientHandler.finalizeSdkHttpFullRequest(BaseClientHandler.java:66)
software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.doExecute(BaseSyncClientHandler.java:151)
software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.lambda$execute$1(BaseSyncClientHandler.java:82)
software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.measureApiCallSuccess(BaseSyncClientHandler.java:175)
software.amazon.awssdk.core.internal.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:76)
software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:45)
software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:56)
software.amazon.awssdk.services.lambda.DefaultLambdaClient.updateFunctionCode(DefaultLambdaClient.java:4804)
magenta.tasks.UpdateS3Lambda.$anonfun$execute$7(tasks.scala:263)
magenta.tasks.UpdateS3Lambda.$anonfun$execute$7$adapted(tasks.scala:252)
magenta.package$.withResource(package.scala:41)
magenta.tasks.Lambda$.withLambdaClient(AWS.scala:110)
magenta.tasks.UpdateS3Lambda.execute(tasks.scala:252)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$4(TasksRunner.scala:37)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$4$adapted(TasksRunner.scala:36)
magenta.DeployReporter$.$anonfun$sendContext$1(logging.scala:121)
magenta.DeployReporter$.withFailureHandling(logging.scala:100)
magenta.DeployReporter$.magenta$DeployReporter$$sendContext(logging.scala:120)
magenta.DeployReporter.taskContext(logging.scala:37)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$2(TasksRunner.scala:36)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$2$adapted(TasksRunner.scala:29)
scala.collection.immutable.List.foreach(List.scala:333)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$1(TasksRunner.scala:29)
deployment.actors.TasksRunner$$anonfun$receive$1.$anonfun$applyOrElse$1$adapted(TasksRunner.scala:27)
magenta.DeployReporter$.$anonfun$sendContext$1(logging.scala:121)
magenta.DeployReporter$.withFailureHandling(logging.scala:100)
magenta.DeployReporter$.magenta$DeployReporter$$sendContext(logging.scala:120)
magenta.DeployReporter.infoContext(logging.scala:42)
deployment.actors.TasksRunner$$anonfun$receive$1.applyOrElse(TasksRunner.scala:27)
akka.actor.Actor.aroundReceive(Actor.scala:537)
akka.actor.Actor.aroundReceive$(Actor.scala:535)
deployment.actors.TasksRunner.aroundReceive(TasksRunner.scala:14)
akka.actor.ActorCell.receiveMessage(ActorCell.scala:580)
akka.actor.ActorCell.invoke(ActorCell.scala:548)
akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
akka.dispatch.Mailbox.run(Mailbox.scala:231)
akka.dispatch.Mailbox.exec(Mailbox.scala:243)
java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```
